### PR TITLE
graalvm-ce: use NATIVE_IMAGE_DEPRECATED_BUILDER_SANITATION instead of CC wrapper

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

Instead of using a hacky CC wrapper to re-inject the environment variables needed to build, we set the `NATIVE_IMAGE_DEPRECATED_BUILDER_SANITATION=true` that is in 21.0.0 release changelog:
https://github.com/oracle/graal/blob/489611a304ec808cdcb1e07e9a9799e6240fd2e1/substratevm/CHANGELOG.md?plain=1#L48

This is deprecated behavior for now, and will be removed in future, but this will allow us more time until we figure a proper solution.

See:
- https://github.com/oracle/graal/issues/7502
- https://github.com/oracle/graal/issues/7981
- https://github.com/oracle/graal/pull/6095

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by thiagokokada** [comment]
@ofborg build babashka clojure-lsp cljfmt clj-kondo jet zprint

**Comment by thiagokokada** [comment]
CC @ericdallo @jlesquembre.

**Comment by jlesquembre** [comment]
Result of `nixpkgs-review pr 273961` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>babashka</li>
    <li>babashka-unwrapped</li>
    <li>bbin</li>
    <li>clj-kondo</li>
    <li>cljfmt</li>
    <li>clojure-lsp</li>
    <li>dapl-native</li>
    <li>dbqn-native</li>
    <li>graalvm-ce (graalvmCEPackages.graalvm-ce)</li>
    <li>graalvmCEPackages.graalvm-ce-musl</li>
    <li>jet</li>
    <li>neil</li>
    <li>scala-update</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
    <li>zprint</li>
  </ul>
</details>

File: pkgs/build-support/build-graalvm-native-image/default.nix [hunk 1]
```diff
@@ -49,6 +49,11 @@ stdenv.mkDerivation ({
 
   nativeImageBuildArgs = nativeImageBuildArgs ++ extraNativeImageBuildArgs ++ [ graalvmXmx ];
 
+  # Workaround GraalVM issue where the builder does not have access to the
+  # environment variables since 21.0.0
+  # https://github.com/oracle/graal/pull/6095
```
Comment (jlesquembre) [inline pkgs/build-support/build-graalvm-native-image/default.nix:54]:
```suggestion
  # https://github.com/oracle/graal/pull/6095
  # https://github.com/oracle/graal/issues/7502
```
I'd add a link to the issue you created, I think the information there is useful

**Review by jlesquembre — APPROVED** [review]
@thiagokokada thanks, I agree in that we need a better solution, but as a temporary workaround is good.
I left a comment about adding more context in the comments, other than that, it looks good to me.

**Comment by thiagokokada** [comment]
@ofborg build babashka clojure-lsp cljfmt clj-kondo jet zprint

**Pull request merged**